repose-aggregator/components/filters/rate-limiting-filter/src/main/java/org/openrepose/filters/ratelimiting/RateLimitingHandler.java

```diff
@@ -109,29 +115,76 @@ private boolean requestHasExpectedHeaders(HttpServletRequest request) {
         return request.getHeader(PowerApiHeader.USER) != null;
     }
 
+    private static Optional<MediaType> parseMediaType(String type) {
+        Optional<MediaType> parsedType = Optional.empty();
+        try {
+            parsedType = Optional.of(MediaType.parseMediaType(type));
+        } catch (IllegalArgumentException iae) {
+            LOG.warn("Media type could not be parsed: {}", type, iae);
+        }
+        return parsedType;
+    }
+
+    private Optional<MediaType> getPreferredMediaType(List<String> acceptValues) {
+        Optional<MediaType> preferredMediaType = Optional.of(DEFAULT_MEDIA_TYPE);
+        if (!acceptValues.isEmpty()) {
+            /*
+            Parses media types from the "Accept" header (dropping any that cannot be parsed),
+            removes any media types that are not supported by this filter (since they will not be returned anyway),
+            then sorts the remaining media types by specificity (primarily) and quality (secondarily).
+              */
+            List<MediaType> sortedRelevantAcceptTypes = acceptValues.stream()
+                    .map(RateLimitingHandler::parseMediaType)
+                    .filter(Optional::isPresent)
+                    .map(Optional::get)
+                    .filter(mediaType -> SUPPORTED_MEDIA_TYPES.stream().anyMatch(mediaType::isCompatibleWith))
+                    .sorted(new CompoundComparator<>(MediaType.SPECIFICITY_COMPARATOR, MediaType.QUALITY_VALUE_COMPARATOR))
```

Comment:
Derp. The `MediaType` comparator is a constant defined as:
`new SpecificityComparator<MediaType>() {
...
}`

In other words, it's an anonymous class that extends the `MimeType` `Comparator`. It doesn't override the `compare` method, only the `compareParameters` method. Then the overridden method calls through in some cases.